### PR TITLE
Fix compile error on JDK15: Use direct AlgorithmId.get()

### LINKED ISSUE #10478: compile error with JDK15
### JDK

```
$ java --version
openjdk 15 2020-09-15
OpenJDK Runtime Environment (build 15+36-1562)
OpenJDK 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)
```

### Expected Behavior

compile success

### Actual Behavior

```
[error] /Users/kenji/temp/playframework/transport/server/play-server/src/main/scala/play/core/server/ssl/CertificateGenerator.scala:43:19: value sha256WithRSAEncryption_oid is not a member of object sun.security.x509.AlgorithmId
[error]       AlgorithmId.sha256WithRSAEncryption_oid
[error]                   ^
[error] /Users/kenji/temp/playframework/transport/server/play-server/src/main/scala/play/core/server/ssl/CertificateGenerator.scala:64:19: value sha256WithRSAEncryption_oid is not a member of object sun.security.x509.AlgorithmId
[error]       AlgorithmId.sha256WithRSAEncryption_oid
[error]                   ^
[error] /Users/kenji/temp/playframework/transport/server/play-server/src/main/scala/play/core/server/ssl/CertificateGenerator.scala:90:93: value md5WithRSAEncryption_oid is not a member of object sun.security.x509.AlgorithmId
[error]     generateCertificate(dn, pair, Date.from(from), Date.from(to), "MD5WithRSA", AlgorithmId.md5WithRSAEncryption_oid)
[error]                                                                                             ^
[error] three errors found
```

- https://github.com/playframework/playframework/blob/810fca22fb1a5b8111fcf84d524349f9a66c22ab/transport/server/play-server/src/main/scala/play/core/server/ssl/CertificateGenerator.scala
- https://github.com/openjdk/jdk/commit/080b3b83ebffe5149fbc9ac48e921fb5

`sun.security.x509.AlgorithmId.sha256WithRSAEncryption_oid` and `sun.security.x509.AlgorithmId.md5WithRSAEncryption_oid` removed(?) since JDK15

```
Welcome to Scala 2.13.3 (OpenJDK 64-Bit Server VM, Java 14).
Type in expressions for evaluation. Or try :help.

scala> sun.security.x509.AlgorithmId.md5WithRSAEncryption_oid
val res0: sun.security.util.ObjectIdentifier = 1.2.840.113549.1.1.4
```

```
Welcome to Scala 2.13.3 (OpenJDK 64-Bit Server VM, Java 15).
Type in expressions for evaluation. Or try :help.

scala> sun.security.x509.AlgorithmId.md5WithRSAEncryption_oid
                                     ^
       error: value md5WithRSAEncryption_oid is not a member of object sun.security.x509.AlgorithmId
```

### Reproducible Test Case

- checkout playframework/playframework
- `sbt compile` with JDK15

### PR DESCRIPTION
**Pull request by mkurz**
Fixes #10478
Simply applied the same fix like https://github.com/lightbend/ssl-config/pull/269

I can compile Play with Java 16 release candiate now!

### CODE REVIEW

File: project/BuildSettings.scala [hunk 1]
```diff
@@ -304,6 +304,9 @@ object BuildSettings {
         "play.api.routing.sird.PathBindableExtractors.play$api$routing$sird$PathBindableExtractors$_setter_$uuid_="
       ),
       ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.routing.sird.PathBindableExtractors.uuid"),
+      // Fix compile error on JDK15: Use direct AlgorithmId.get()
+      ProblemFilters
+        .exclude[IncompatibleMethTypeProblem]("play.core.server.ssl.CertificateGenerator.generateCertificate"),
```
Comment (mkurz) [inline project/BuildSettings.scala:309]: Wondering why MiMa complains, `generateCertificate` is defined as `private[play]`...

**Comment by mkurz** [comment]
@ignasi35 or @octonato This should be an easy one to review :wink:

**Comment by mkurz** [comment]
@raboof @ignasi35 Can you review? Should be easy, its the same like https://github.com/lightbend/ssl-config/pull/269

File: transport/server/play-server/src/main/scala/play/core/server/ssl/CertificateGenerator.scala [hunk 2]
```diff
@@ -87,7 +86,7 @@ object CertificateGenerator {
     val keyGen = KeyPairGenerator.getInstance("RSA")
     keyGen.initialize(keySize, new SecureRandom())
     val pair = keyGen.generateKeyPair()
-    generateCertificate(dn, pair, Date.from(from), Date.from(to), "MD5WithRSA", AlgorithmId.md5WithRSAEncryption_oid)
+    generateCertificate(dn, pair, Date.from(from), Date.from(to), "MD5WithRSA", AlgorithmId.get("MD5WithRSA"))
```
Comment (mkurz) [inline transport/server/play-server/src/main/scala/play/core/server/ssl/CertificateGenerator.scala:89]: `MD5WithRSA` is defined here:
https://github.com/openjdk/jdk8u/blob/jdk8u292-b01/jdk/src/share/classes/sun/security/x509/AlgorithmId.java#L960
https://github.com/openjdk/jdk8u/blob/jdk8u292-b01/jdk/src/share/classes/sun/security/x509/AlgorithmId.java#L545-L547

**Review by raboof — APPROVED** [review]

**Comment by mkurz** [comment]
Thanks @raboof!

**Pull request merged**